services/self-service/src/main/java/com/epam/dlab/backendapi/service/impl/UserGroupServiceImpl.java

```diff
@@ -83,6 +88,12 @@ public void removeGroupFromRole(Set<String> groups, Set<String> roleIds) {
 
 	@Override
 	public void removeGroup(String groupId) {
+		if (projectDAO.getProjectsWithStatusNotIn(ProjectDTO.Status.DELETED, ProjectDTO.Status.DELETING).stream()
+				.anyMatch(p -> p.getGroups().contains(groupId) && p.getGroups().size() == 1)) {
```

Comment:
It is better to forbid removing group if it is used in any of project (does not matter if p.getGroups().size() == 1) 
Admin should go and remove such group manually from project